### ISSUE TITLE
Configuration Data not saving in EEPROM

### ISSUE THREAD

**Issue by SenthilKumar8248** [bigtreetech/BIGTREETECH-SKR-E3-Turbo#16]
I have flashed new marlin firmware 2.0.7.2 with SKR E3 Turbo which is issued by Bigtreetech. If I set new temperature value in "Preheat PLA config" and pressed "store setting". And after restart the Ender-3, the Temperature setting show default value. The value which I stored is not retrieving from EEPROM. Please advise.

**Comment by Xtreembass** [comment]
Hi.
I have the same issure, when i usesed a self compiled FW.
When:
Send: M500
Recv: echo:Error writing to EEPROM!
Recv: Error:Field esteppers mismatch.
WARNING! Received an error from the printer's firmware, ignoring that as configured but you might want to investigate what happened here! Error: Field esteppers mismatch.
Recv: echo:Error writing to EEPROM!
When i use the Orginal BIN i can safe to the EEprom
Still 3 Days i try to figure it out.
Maybe someone can help us here
Thanks a lot

**Comment by SenthilKumar8248** [comment]
You are absolutely right. I have the same issue. See screen shot. M500 shows Error writing to eeprom. ![image](https://user-images.githubusercontent.com/76647528/103169722-61d1c900-4864-11eb-9ff0-90982a7578a3.jpeg)

![image](https://user-images.githubusercontent.com/76647528/103169727-6eeeb800-4864-11eb-84b9-c2458ecf2b9c.jpeg)

**Comment by Xtreembass** [comment]
I also actived in the Firmware: #define DEBUG_EEPROM_READWRITE so i can see what is wrong

**Comment by trengtor** [comment]
> I also actived in the Firmware: #define DEBUG_EEPROM_READWRITE so i can see what is wrong

![2020-12-27_16-42-22](https://user-images.githubusercontent.com/16775489/103172159-8e84e100-4862-11eb-9fe9-6ecad0b97614.jpg)

**Comment by Xtreembass** [comment]
The setting is not only uncomment something, type it in. 
it will work

**Comment by SenthilKumar8248** [comment]
> I also actived in the Firmware: #define DEBUG_EEPROM_READWRITE so i can see what is wrong

Where is the line of code exist. Whether it's in configuration.h or configuration_adv.h

**Comment by Xtreembass** [comment]
i put that in Configuration.h in the last line, fill it in there

**Comment by rickd1994** [comment]
i have the same problem eprom does not save

**Comment by Xtreembass** [comment]
i try the new bugfix version from yesterday on my testboard, and it works, tomorow i try it on my printer

**Comment by SenthilKumar8248** [comment]
> i try the new bugfix version from yesterday on my testboard, and it works, tomorow i try it on my printer
> 
> 

Yes! After using bugfix version, now I can able to save in eeprom.

**Comment by calvinmclean** [comment]
Where do you find this bugfix version? Is that from the [Marlin repo](https://github.com/MarlinFirmware/Marlin/tree/bugfix-2.0.x)? I tried with that branch but ran into some issues with configuration. Not sure if I should be looking somewhere else or just try harder to make the new config changes

**Comment by Xtreembass** [comment]
Yes the bugfix from the marlin Homepage, but you have to reload the config from the examples (also on the Homepage).
The bugfix config is differnt to the "normal" Version

**Comment by calvinmclean** [comment]
This worked for me as well. Thanks for the help

**Comment by rickd1994** [comment]
the bugfix version also works for me thx for the help

**Comment by SenthilKumar8248** [comment]
Hi, after using bugfix version, I have new issue "Heating failed: E1 printer halted please restart". I have completed PID Autotune and saved PID value in eeprom, but still getting this issue when I preheat the extruded.
![Uploading 103457803...]

**Comment by Xtreembass** [comment]
Detto on my side, also my steppers run too hot, try to lower the Current, now i am at 630 and i run test number : i dont know ;)

**Comment by SenthilKumar8248** [comment]
> Detto on my side, also my steppers run too hot, try to lower the Current, now i am at 630 and i run test number : i dont know ;)

Follow the steps shown in this video. Now I have no Heating issue.

https://youtu.be/ZwosxcGmT_Y

**Comment by dk83** [comment]
I have the same mistake. No data is saved in the EEPROM.
If the bugfix version is used, "#define EEPROM_AUTO_INIT" is active. 
Then no error message appears on the display - but no values ​​are saved either.

In spite of this, the terminal reports that the EEPROM cannot be initialized.

**Please fix the problem.**

The only interim solution:
Open the file pins_BTT_SKR_E3_TURBO.h.
Then deactivate I2C EEPROM and activate VIRTUAL_SD_EEPROM.
E.g.:
// Onboard I2C EEPROM
// # define I2C_EEPROM
// # define MARLIN_EEPROM_SIZE 0x1000 // 4KB (AT24C32)

// Use Virtual SDCARD eeprom
#define VIRTUAL_SD_EEPROM 8

**Comment by Xtreembass** [comment]
The problem is fixed in the last bugfix version

**Comment by baikal** [comment]
There is a simple fix if one does not want to use the latest bugfix-2.0.x from Marlin, but e.g 'Marlin-2.0.7.2-SKR-E3-Turbo' from this repo:
Remove or comment line ~51 from **Marlin/src/HAL/LPC1768/HAL.h** which defines a bogus I2C_ADDRESS macro
`
//#define I2C_ADDRESS(A) uint8_t((A) << 1)
`
Then, the i2c EEPROM works just fine.
I guess that what bigtreetech must have done to make the .bin in this repo, as they do work.

**Comment by dk83** [comment]
//#define I2C_ADDRESS(A) uint8_t((A) << 1)

That works perfectly for me

**Comment by TheNeoShogun** [comment]
It solve my problem with EEPROM too.
